content/en/blog/2023/traffic-for-ambient-and-sidecar/index.md

```diff
@@ -0,0 +1,105 @@
+- httpbin deployed in namespace bar-1
+    - httpbin pod is scheduled to Node B
+    - the waypoint proxy of httpbin is disable
+- helloworld is deployed in namespace bar-2
+    - helloworld pod is scheduled to Node D
+    - the waypoint proxy of helloworld is enabled
+    - the waypoint proxy is scheduled to Node C
+- foo namespace enables sidecar injection (foo namespace contains label: `istio-injection: enabled`)
+- bar-1 namespace enables ambient mode (bar-1 namespace contains label: `istio.io/dataplane-mode=ambient`)
+
+With the above description, the deployment and network traffic paths are:
+
+{{< image width="100%"
+    link="sidecar-to-ambient.png"
+    caption="sleep to httpbin and helloworld"
+    >}}
+
+### [ sleep -> sidecar ] -> ztunnel -> httpbin network traffic path
+
+Based on the above deployment, the first part is shown on top half of the diagram: the network traffic path of the sleep pod in sidecar mode to the httpbin pod in ambient mode.
+
+**(1) (2) (3) (4)** the sleep container sends a request to httpbin. The request is intercepted by iptables rules and directed to port 15001 of the sleep pod sidecar. Then, the sidecar handles the request and routing rules by following configurations distributed by the control plane. Next, the request traffic is forwarded to a specific httpbin pod in node B, identified by its ip address.
+
+**(5) (6)**  After the request is sent to the device pair (`veth httpbin <-> eth0 inside httpbin pod`), the request would be intercepted and forwarded to the `istioin` device on the node B where httpbin pod is running by following its iptables and route rules. The `istioin` device in the node B is a `[geneve](https://www.rfc-editor.org/rfc/rfc8926.html)` tunnel, and the other side of this tunnel is `pistioin` device which is inside the ztunnel pod of node B.
+
+**(7) (8)** After the request enters the `pistioin` device of the ztunnel pod, the request is intercepted and redirected through port 15008 of the ztunnel pod by the iptables rules of the ztunnel pod on node B.
+
+**(9)** The traffic getting into the port 15008 would be considered as a inbound request, then ztunnel will forward the request to the httpbin pod in the same node B.
+
+### [ sleep-> sidecar ] -> waypoint -> ztunnel -> helloworld httpbin network traffic path
+
+Comparing with the first part of the diagram, it's clear that the waypoint proxy is added in path **"[ sleep -> sidecar ] -> ztunnel -> httpbin"**. The Istio control plane has all information of service and configuration of the service mesh. When helloworld is configured with a waypoint proxy, the EDS configuration of helloworld service received by sidecar of sleep pod will be changed to the type of `envoy_internal_address`. This causes that the request traffic going through the sidecar to be forwarded to port 15008 of the waypoint proxy on node C via the `[HBONE](https://docs.google.com/document/d/1Ofqtxqzk-c_wn0EgAXjaJXDHB9KhDuLe-W3YGG67Y8g/edit)` protocol.
```

Comment:
given this is a deep dive, i'd expect a snippet of waypoint config here.

Reply:
Hi @linsun, as you know, the situation is changing so such detail information may be outdated very soon. How about to just mention the basic implementation?